addons/account/models/sequence_mixin.py

```diff
@@ -217,11 +224,20 @@ def _get_last_sequence(self, relaxed=False, with_prefix=None, lock=True):
             param['with_prefix'] = with_prefix
 
         query = f"""
-                SELECT {{field}} FROM {self._table}
+            WITH filtered_recs AS (
```

Comment:
I'm not sure how you made your profiling, but here is the `EXPLAIN ANALYZE` I got on real data.
Your version: 256ms https://explain.dalibo.com/plan/8470f5a45cgg644a
The current version: 0.73ms (including the planning time!) https://explain.dalibo.com/plan/1a2e71g3c19a4722

One way to fix your version would be to use `NOT MATERIALIZED` in the CTE declaration, but then it is exactly the same thing as before, only with a different syntax. (syntax is arguably better, but not worth the diff)

Reply:
I see you're not taking into account when the method is called with `with_prefix=True`. Please could you benchmark with that enabled?

My benchmarks:
- Before: https://explain.dalibo.com/plan/f2da5d19257643bf (Execution time: 29,4ms, Planning time: 1,85ms)
- After the patch: https://explain.dalibo.com/plan/c82946f754a9563e (Execution time: 16,5ms, Planning time: 1,58ms)

Keep in mind that the amount of data can alter the plan Postgres does. In my case, the `account_move` table has ~650000 records.

Reply:
> I'm not sure how you made your profiling, but here is the `EXPLAIN ANALYZE` I got on real data.

Make sure you upload results using `EXPLAIN (ANALYZE, COSTS, VERBOSE, BUFFERS, FORMAT JSON)`, for best results.

Reply:
> I see you're not taking into account when the method is called with `with_prefix=True`. Please could you benchmark with that enabled?

It doesn't matter.
If the CTE is materialized, and there are a lot of entries in the same sequence, Postgres will need to build a big "table", without indexes, leading to 2 seq scans of it to retrieve the `ORDER BY .. LIMIT 1` instead of a almost `O(1)` index scan (there shouldn't be too many rows to discard during the index scan)
If it is not materialized, then the plan will be exactly the same.

Also, according to your benchmarks, you can only get a **small** improvement in some cases.
According to my benchmarks, you can also get **huge** diminishments.


> Make sure you upload results using `EXPLAIN (ANALYZE, COSTS, VERBOSE, BUFFERS, FORMAT JSON)`, for best results.

I know how to do an explain :slightly_smiling_face: 
Note that `COSTS` default to true. I already turned on VERBOSE, and I prefer the text format because it is easier to copy/paste and read with almost as much information.

Reply:
> I prefer the text format because it is easier to copy/paste and read with almost as much information.

In my computer, copy/pasting costs exactly the same... That makes the browsable version harder to understand, like in the totals:
![image](https://github.com/user-attachments/assets/784ef1ea-bf45-4d08-960f-1a3f40ecc690)

I guess it's not the most important detail.

> It doesn't matter.

Well, there *must* be a difference somewhere, because otherwise benchmarks would lead in the same direction, don't you think?

So, if it isn't the obvious fact that we're benchmarking different queries, then what justifies that difference?

And, most important, do you have any suggestions on how to improve performance here? I've spent 12 hours on it but couldn't get to something extremely significant (if better). 🤔 

However performance in this part of the code is still very bad. 🤦🏼‍♂️  There must be something that can be done...

Reply:
> Well, there must be a difference somewhere, because otherwise benchmarks would lead in the same direction, don't you think?

You might have a positive difference only because your sub-sequence is really small and therefore not including many moves; the only case where materializing the CTE is good.

> And, most important, do you have any suggestions on how to improve performance here?

If I had I would probably already have done it :wink: 
This part takes time because it needs to be done one by one, because we rely on the SQL unique constraint on the name to assign the next number.

We could try batching the writes, but it is not straightforward at all.

> However performance in this part of the code is still very bad. 🤦🏼‍♂️ There must be something that can be done...

I'm currently looking into reducing time of the CI. That includes a `populate` build, which includes mass posting. Will see what I'll find!

Reply:
> This part takes time because it needs to be done one by one, because we rely on the SQL unique constraint on the name to assign the next number.

I'm thinking something. Maybe if we can know beforehand which move ids will be updated, we can have a method that assigns the number to all of them in a single query. Once that's done, the rest of the code should be able to continue normally without depending on manual flushes. Makes sense?